### PR TITLE
Restrict tigera-operator secret access to namespace only

### PR DESCRIPTION
**Pull request by vara2504**
Restrict tigera-operator secret access to namespace only, retain get/list/watch cluster-wide 
https://tigera.atlassian.net/browse/EV-5410

Tested external prometheus ,
bz addons run external-prometheus:setup
bz addons run wireguard:setup
image

External-prometheus E2E  passed

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by rene-dekker — APPROVED** [review]
lgtm

File: pkg/render/monitor/monitor.go
Comment (rene-dekker) [inline pkg/render/monitor/monitor.go:189]: On line 239 we may be creating a secret for which the operator requires permissions. Should we render a binding in that namespace prior to rendering the secret?

**Review by rene-dekker — CHANGES_REQUESTED** [review]
See the comment in monitor.go

**Review by rene-dekker — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
